### DISCUSSION TITLE
在FAKE-IP增强模式下，PS5nat类型失败，求大佬们解答

### DISCUSSION THREAD

**Discussion by KiopPsFwMy** [vernesong/OpenClash#3939] [Q&A]
如标题所示，在Fake-IP增强模式开启UDP转发的情况下，PS5一直nat失败，PSN商店时不时就断线
通过的连接都是DIRECT直连.

**Comment by 19901019** [comment]


**Reply by KiopPsFwMy** [reply]
感谢解答，但我这边面板看见stun.playstation.net的类型就是UDP
今天又鼓捣了一阵
看见有人说是因为网络通过的设备太多了，所以ps5才nat失败，改天把我这旁路游换到主路由试试

**Comment by hunkding** [comment]
我目前也是识别不了NAT类型

**Reply by d808b** [reply]
把[stun.playstation.net]加入直通名单，显示NAT3,不再显示失败！

**Comment by ytq77** [comment]
我改成TUN混合,覆写-DNS里面把fakefilter加上了stun  依旧是nat失败  估计应该还要在防火墙里面开什么端口之类的吧   目前我也没搞定

**Reply by eedaniel12** [reply]
我发现PS流量只要通过openclash核心就会nat失败，我现在的方法是openclash绕过PS，所有流量不走核心，然后路由器开UU加速，UU加速里面设置全平台加速，完美解决问题，只是需要开个UU会员，好在我打游戏本身就需要加速器

**Reply by ytq77** [reply]
我是想用ps remote play来着,正常开加速器指定直连是nat没问题的....就是走代理的时候没法实现这个nat稳定   我还在整理思路  开放了几个端口  还把相关的域名都加到了返回真实ip的黑名单  还没测试效果

**Reply by ytq77** [reply]
破案了  关了openclash就是没事   开着不行是因为我的节点不支持udp  我得去搞个支持udp的才能继续尝试

**Reply by seven11sys** [reply]
> 我发现PS流量只要通过openclash核心就会nat失败，我现在的方法是openclash绕过PS，所有流量不走核心，然后路由器开UU加速，UU加速里面设置全平台加速，完美解决问题，只是需要开个UU会员，好在我打游戏本身就需要加速器

你好，我也是一样的情况，一直在用uu，请问应该怎么设置让openclash绕过PS？

**Reply by ytq77** [reply]
<img width="2332" height="381" alt="image" src="https://github.com/user-attachments/assets/9afe8802-8fc1-40b0-8285-e677f429a76c" />
插件设置最下面有一个这个,设置一个ip的所有端口绕过就行,这样的话这个设备不想走内核的时候就改成这个ip就好了

**Reply by seven11sys** [reply]
> <img alt="image" width="2000" height="381" src="https://private-user-images.githubusercontent.com/61270275/476681105-9afe8802-8fc1-40b0-8285-e677f429a76c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUxNzUwNDksIm5iZiI6MTc1NTE3NDc0OSwicGF0aCI6Ii82MTI3MDI3NS80NzY2ODExMDUtOWFmZTg4MDItOGZjMS00MGIwLTgyODUtZTY3N2Y0MjlhNzZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODE0VDEyMzIyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRkZDZhZGEyNDI2NDJjM2ExNTY2MDFmYzlhY2Y1OWRiOTdkZWYyMzFiY2QxNWY3Y2VkYWZkZjJhMDg1Y2M3NjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hbR9-p4L9Zeh4NW1GzIIw0DoouIx4D2mDgolsBnhiBE"> 插件设置最下面有一个这个,设置一个ip的所有端口绕过就行,这样的话这个设备不想走内核的时候就改成这个ip就好了

好的，谢谢

**Comment by 19901019** [comment]


**Reply by ytq77** [reply]
看得懂倒是,参考你这个刚好发现配置文件一个地方的顺序不太对,调换了一下.

我现在就是指定了一个ip走'来源流量访问控制',这样这个ip不走核心,可以正常使用ps remote play(至少达成了一半的目标

但是走过墙的节点依旧是nat失败  正常我的网络环境是nat2,也无法通过ps remote 来链接,这个目前还没找到解决办法.
看起来你的nat走的是主机联机,我现在跟你差不多,我是在覆写里面直接指定了SCR-IP,主机ip,单独的游戏节点,看后台stun的域名也走的这个节点
感觉问题就是出在节点上了,UPNP之类的我也做了,也可以正常打开udp端口

顺便贴一下我的两种配置文件
https://raw.githubusercontent.com/yixuan-ovo/TutorialFiles_yx/refs/heads/main/MihomoParty/subscribe_ini/yx_hxy_all.ini
https://raw.githubusercontent.com/yixuan-ovo/TutorialFiles_yx/refs/heads/main/OpenClash/subscribe_ini/yx_clash.ini

**Reply by ytq77** [reply]
这个的用法是覆写里面SCR-IP,主机ip,PS单用,翻墙用的

![image](https://github.com/user-attachments/assets/1be97944-1b52-49db-90dc-55d34b94b138)
我的openclash用的是第二个配置文件,刚才看了你的我又去翻了一遍规则,果然发现stun这个域名在另一个规则里面写了,所以我把那个规则也加了UDP节点了

顺便请教一下有没有测试节点支持不支持UDP的方法

**Reply by ytq77** [reply]
![image](https://github.com/user-attachments/assets/1fc03187-6162-4c22-bf80-10dddc2da974)

**Reply by ytq77** [reply]
规则外走全局直连吗?还是全局走节点

**Reply by ytq77** [reply]
我现在就是这么做的  也不行 我也是无语了

**Reply by ytq77** [reply]
试了  不行啊老哥,我之前在这里设置了跟你差不多意思的走指定节点

![image](https://github.com/user-attachments/assets/34fd3bda-8331-4507-a940-9d0e8493a9ce)

**Reply by ytq77** [reply]
嗯,我还没试关闭绕过大陆,等着尝试一下

**Comment by sniper3209** [comment]
经过我的实测，发现在openclash中把禁用 quic-go GSO 支持打开就行了，不用去动别的设置

**Reply by ytq77** [reply]
试了一下  还是不稳定  看看你的其他配置可以不

**Reply by sniper3209** [reply]
确实，我现在也发现不是很稳定，目前我还没有解决

**Comment by sniper3209** [comment]
目前我是主路由+旁路由的拓扑：
主开启拨号+DHCP+UPNP，旁只负责分流（openclash）。
之前一直没有注意的是我两只路由都开启了UPNP,
但同一网络中只应有一个 UPnP 服务，否则可能导致设备识别混乱！
所以，我把旁的UPNP关了，然后把stun.playstation.net直连添加到了自定义规则里，目前都是秒测NAT2!
还有就是规则的订阅，我推荐一个大佬的WIKI供大家参考：
https://github.com/Aethersailor/Custom_OpenClash_Rules/wiki/OpenClash-%E8%AE%BE%E7%BD%AE%E6%96%B9%E6%A1%88
祝联机愉快

**Reply by sniper3209** [reply]
打脸来的太快了，刚发完帖就失败了，是不是有一种可能，这就是运营商的锅？

**Reply by ytq77** [reply]
这个,,,一般刚更新完的几分钟  大概五分钟内吧,或者三四次  nat都是会稳定,再过会就寄了的.....

**Reply by ytq77** [reply]
但是我感觉不是运营商的锅,我上面设置了ps5绕过内核的话,是可以正常nat和ps remote远程游玩的,所以应该是openclash的问题
我主要是为了能正常使用ps remote,如果在openclash环境下,远程会提示8801330d还是什么的一个错误码
绕过op内核之后就一切正常了

**Reply by d808b** [reply]
投降吧！換用openwrt-passwall2，OpenClash太折騰了！現在PS5、蘋果全家桶正常！

**Reply by ytq77** [reply]
惆怅呀...

**Reply by sniper3209** [reply]
遗憾呀...

**Comment by sniper3209** [comment]
我想我目前应该是成功了！两天时间，不定时间隔测定NAT类型，均是2！
首先：目前我是主路由+旁路由的拓扑：主开启拨号+DHCP+UPNP，旁只负责分流（openclash）。
然后需要把旁路由的UPNP关了，然后**stun.playstation.net还是要走节点!**
还有就是规则的订阅，我就是用的这个，供大家参考：
https://github.com/Aethersailor/Custom_OpenClash_Rules/wiki/OpenClash-%E8%AE%BE%E7%BD%AE%E6%96%B9%E6%A1%88
再就是旁路由详细设置：
一、关闭旁路由Upnp服务功能
二、在 旁路由 网络-防火墙 选项中
1、不要勾选 启用 SYN-flood 防御
2、关闭路由/NAT 卸载 ---流量卸载类型（无）
3、入站数据、出站数据及转发 均选择 接受
4、勾选 IP动态伪装
5、勾选 开启 FullCone NAT
6、在流量卸载类型中，选 无
openclash设置：
一、 fake ip 增强模式
二、勾选 UDP 流量转发
二、覆写设置-DNS设置 中
禁用“追加上游 DNS”，在 NameServer 中保留第一个服务器，并将地址修改为你的运营商 DNS 地址。同时，禁用 Fallback 和 Default-NameServer 下的所有服务器并勾选 遵循规则(respect-rules)
三、然后是在 配置管理 中修改Yaml文件
在配置文件（左侧）的顶层结构中添加以下字段 （建议放在 mode 或 log-level 之后）： disable-udp: false
具体位置示例如下：
port: 7890
socks-port: 7891
allow-lan: true
mode: rule
log-level: info
**disable-udp: false**
external-controller: 127.0.0.1:9090
请仔细检查你的yaml是否有 disable-udp: false字段，没有在话就在照示例的位置加进去，记得保存及应用。
但是这里要注意一个细节，不要自动定时更新订阅，因为是后期修改的，所以一但更新，你还需要手动再修改一下！
四、如果有观看YouTube的需求，则要插件设置-流量控制中 把禁用quic的选项勾选，确保速度正常，然后来源流量访问控制中，新增你PS5的内容（见图）
<img width="2097" height="455" alt="image" src="https://github.com/user-attachments/assets/a79a7c60-1d8f-4640-a697-92592050d4a9" />
这样就可以大功告成了！祝大家用机愉快

**Reply by sniper3209** [reply]
MHR联机和NAT类型，随机挑时间测了三次，NAT2,MHR联机正常！
将保持持续随机测试至周末

**Reply by sniper3209** [reply]
今天测试正常

**Reply by ytq77** [reply]
建议在非局域网环境下使用ps remote远程连接,看看是否可以连接上

**Reply by sniper3209** [reply]
> 建议在非局域网环境下使用ps remote远程连接,看看是否可以连接上
我在单位用手机ps remote远程连接，没有问题
刚才又测了一次，NAT2,正常

**Reply by sniper3209** [reply]
今日测试正常

**Reply by ytq77** [reply]
我没有旁路由,我看应该就是开TUN,然后禁用追加上游DNS,勾选遵循规则,但是disable-udp这个应该在配置文件里面有了吧?

**Reply by sniper3209** [reply]
> 我没有旁路由,我看应该就是开TUN,然后禁用追加上游DNS,勾选遵循规则,但是disable-udp这个应该在配置文件里面有了吧?

这个你要去检查一下，我的就没有，我估计你可能也没有，在openclash的 配置管理 中可以看到的

**Reply by sniper3209** [reply]
> 我没有旁路由,我看应该就是开TUN,然后禁用追加上游DNS,勾选遵循规则,但是disable-udp这个应该在配置文件里面有了吧?

你只有主路由的话，设置是和我不同的，请根据我给的大佬设置教程一步步来，你肯定可以成功

**Reply by ytq77** [reply]
我一直用的他的规则模板,自己改了下模板的分组

**Reply by d808b** [reply]
警告⚠️此防火墙配置只适用于「旁路由」不适合「主路由」使用！
# 主路由「常规设置」建议修改为：
1. 启用 SYN-flood 防御 ✓
2. 入站数据 → 拒绝（仅通过端口转发开放必要服务）✓
3. 转发 → 拒绝 ✓
4. 出站 → 接受 ✓
5. 保持 FullCone NAT ✓（如有游戏/P2P需求）
6. IP动态伪装 ✓
7. 流量卸载 → 根据硬件支持选择（如无问题可尝试启用）
<img width="947" height="183" alt="截屏2026-02-07 14 01 57" src="https://github.com/user-attachments/assets/47115792-2ce5-4e3b-8172-2a477d972058" />

**Comment by sniper3209** [comment]
经测试，只要将stun.playstation.net改成直联，还是会有失败的情况

**Reply by ytq77** [reply]
我直接固定一个ip给ps5整个机器走固定节点
没招了

**Reply by sniper3209** [reply]
这也是一个办法，只要成功就行了，每个人的网络条件都不是一样的

**Reply by sniper3209** [reply]
今日测试正常！

**Reply by sniper3209** [reply]
今日测试正常！按目前状态，应该确认没有问题了！结束

**Comment by hunkding** [comment]
在PW规则列表选项内，                                                                                                                                                                                         直连列表添加                                                                                                                                                                                                              #GranTurismo 7直连
stun.*.playstation.net
stun.server.live.playstation.net                                                                                                                                                                                      代理列表添加                                                                                                                                                                                                                #GranTurismo 7代理
*.np.dl.playstation.net
*.playstation.net
*.sonyentertainmentnetwork.com
auth.api.sonyentertainmentnetwork.com
account.sonyentertainmentnetwork.com
*.gran-turismo.com
gt7mob01-gt7tc.ww.np.dl.playstation.net
prod-gt7mob01-gt7tc.ac.np.dl.playstation.net

#sony
*.stun.playstation.net
*.rp.playstation.net
*.playstation.com
*.sony.com
*.api.playstation.com
*.dl.playstation.net
*.turn.playstation.net
time.playstation.com                                                                                                                                                                                                                     即可解决正常联机，开房，入别人房间都正常

**Reply by Noir97** [reply]
Saved my day! 感谢大佬

**Comment by ytq77** [comment]
更新一下,我现在nat类型也正常可以测得2了,因为我还装了一个AGH,我在agh里面禁止了ps5走agh就可以测试成功了,等我明天再测试一下
另外我的openclash里面指定了ps5全部ip走代理,禁止了quic支持.但是ps remote play无法正常连接,还是必须要绕过openclash内核才可以

**Comment by samcg48** [comment]
我折腾了一晚上 Gemini，手搓了几行代码加到 yaml 配置文件解决了……但我不确定现在还有没有用，甚至不知道是不是只对我适用……

 - 
 # 在配置文件的 dns 模块、fake ip filter 下加入这几行域名
 
   fake-ip-filter:
    - "+.playstation.net"
    - "+.playstation.com"
    - "+.sonyentertainmentnetwork.com"
    
 - 
# 拿到 PS5 ip 地址后在 openwrt 中给 PS5 做 IP/MAC 绑定
# 在 rule 模块底下强制让这个 ip 的 udp 流量走直连，ip 字段根据 PS5 的 ip 地址填入


   rules:
    - AND,((SRC-IP-CIDR,192.168.x.x/x),(NETWORK,UDP)),DIRECT
  
  - 
 目前在混合模式、增强模式下都测了，混合模式秒出 nat 2……我也不知道这个方案是否有适用性，但如果有幸能帮到别人，那我会很开心……

**Comment by GleenJi** [comment]
测试失败是因为使用了fakeDNS导致的，对于udp的nat想达到理想的类型2或中等，只能使用real-ip，经过了解和自己摸索后大概就是这样的，如果代理支持fullcone，在一般的家庭网络拓扑中（ps5处于局域网）使用real-ip后ps5能达到的nat类型最高就是类型2了，想达到类型1应该是ps5必须自身能拥有公网ip且没防火墙限制，只要使用了支持fullcone的代理，全程和路由器、光猫什么upnp、dmz设置没任何关系，无需开启这些功能，如果使用直连光猫拥有公网ip且ps5和光猫桥接可能能让ps5达到类型1
